### DISCUSSION TITLE
A toggle.action for Link-Hint and Link-Alphabet(default)

### DISCUSSION THREAD

**Discussion by tryproxy** [gdh1995/vimium-c#717] [Q&A] [answered]
Some pages are easy to navigate with alphabetic hints (those pages usually have little amount of text) and some pages with link-hints(pages with lots of text-links). If you like to visit both such pages, It becomes an inconvenience when you have to compromise one mode for another, since in order to change modes you have to go to the options menu and do it manually.

Unless I've missed "that" `action` in command list, a feature that allows to toggle two modes on the fly will be appreciated. Or perhaps having two separate commands, e.g. `LinkHints.activateABC`, `LinkHints.activateLHint`, to map will also cover this use-case

**Comment by gdh1995** [comment, marked as answer]
You may specify `useFilter=true/false`. For example:
```
map f LinkHints.activate useFilter=false
map <a-f> LinkHints.activate useFilter
```

Or you may use the command named `toggleSwitchTemp`:
```
map <a-s-f> toggleSwitchTemp key="filterLinkHints" $then="f"
```

And then `Alt+Shift+F` will toggle the state of `useFilter` and then start `LinkHints.activate` by `f`. Its effect will keep until a page is closed, reloaded or replaced.

**Reply by tryproxy** [reply]
Thank you for the detailed explanation, didn't guess one can extend commands like that, much appreciated.
